source/gitidentifier.go

```diff
@@ -18,6 +19,8 @@ type GitIdentifier struct {
 	KnownSSHHosts    string
 }
 
+var gitSSHRegex = regexp.MustCompile("^([a-z0-9]+@)?[^:]+:.*$")
```

Comment:
could we make this stricter. user should not be optional (because ambiguous then) and let's validate that between `@` and `:` there is a valid hostname.

Reply:
I think we can, the only potential issue is doing a `git clone host:/path/.../project.git` where we expect it to be using the current user. The regex is requiring a `:` somewhere in the clone path. But I'm fine changing it to require a specific user.

Reply:
current user would be `root` and that is wrong already. Please also validate that `:` is only after hostname and not in a random place in url. And add some unit test cases to cover this.